elasticapm/metrics/base_metrics.py

```diff
@@ -164,6 +167,15 @@ def _metric(self, container, metric_class, name, reset_on_collect, labels):
                     pattern.match(name) for pattern in self._registry._ignore_patterns
                 ):
                     metric = noop_metric
+                elif len(container) >= DISTINCT_LABEL_LIMIT:
```

Comment:
This will allow 1000 counters, 1000 gauges, and 1000 timers. While it means we won't exhaust memory, the logged message is different in meaning. Maybe keep a separate counter?

Reply:
The log message is confusing anyway tbh (why bring up transaction names in a place that has nothing to do with transaction names, just because one of the possible causes for exhausting the limit could be a large cardinality of transaction names?)

I'll do `if len(self._gauges) + len(self._counters) + len(self._timers) >= ...`

Reply:
> why bring up transaction names in a place that has nothing to do with transaction names, just because one of the possible causes for exhausting the limit could be a large cardinality of transaction names?

I suspect this to be the most common source of exhausting the limit, especially if custom metrics are not supported. I think it's valuable to include some indication where the problem might be, even if that's only for 80% of the cases. Arguably, the wording could be improved to better reflect that this is only a potential source of the error.